pandas/io/common.py

```diff
@@ -244,6 +256,66 @@ def file_path_to_url(path):
 }
 
 
+def split_auth_from_url(url_with_uname):
+    """
+    If a url contains username and password, it is extracted and returned
+    along with a url that does not contain it.
+
+    Parameters
+    ----------
+    url_with_uname : a url that may or may not contain username and password
+            see section 3.1 RFC 1738 https://www.ietf.org/rfc/rfc1738.txt
+            //<user>:<password>@<host>:<port>/<url-path>
+
+    Returns
+    -------
+    (username, password), url_no_usrpwd : username or "", password or "", 
+         url without username or password (if it contained it )
+    """
+    o = parse_url(url_with_uname)
+    usrch = '{}:{}@{}'.format(o.username, o.password, o.hostname)
+    url_no_usrpwd = url_with_uname.replace(usrch, o.hostname)
+    return (o.username, o.password), url_no_usrpwd
+
+
+def get_urlopen_args(url_with_uname, auth=None, verify_ssl=True):
+    """
+    generate args to pass to urlopen - including basic auth and and support
+    for disabling verification of SSL certificates ( useful where
+    self-signed SSL certificates are acceptable security risk -eg: Testing )
+
+    Parameters
+    ----------
+    url_with_uname : a url that may or may not contain username and password
+            see section 3.1 RFC 1738 https://www.ietf.org/rfc/rfc1738.txt
+            //<user>:<password>@<host>:<port>/<url-path>
+    auth : ( username/""/None, password/"", None) tuple
+    verify_ssl: If False, SSL certificate verification is disabled.
+
```

Comment:
See my comment <a href="https://github.com/pandas-dev/pandas/pull/16910/files#r127317170">here</a> to patch the formatting for your parameters listed above.